aioshelly/rpc_device/device.py

```diff
@@ -538,3 +545,34 @@ def _parse_dynamic_components(self, components: dict[str, Any]) -> None:
                 for item in self._dynamic_components
             }
         )
+
+    async def _retrieve_blutrv_components(self, components: dict[str, Any]) -> None:
+        """Retrieve BLU TRV components."""
+        if self.model != MODEL_BLU_GATEWAY_GEN3:
+            return
+
+        if not self._config or not self._status:
+            raise NotInitialized
+
+        for component in components.get("components", []):
+            _key = component["key"].split(":")
+            if _key[0] == BLU_TRV_IDENTIFIER:
+                calls = [
+                    ("BluTrv.GetRemoteConfig", {"id": _key[1]}),
+                    ("BluTrv.GetRemoteStatus", {"id": _key[1]}),
+                ]
+                results = await self.call_rpc_multiple(calls)
+
+                cfg: dict[str, Any] = results[0]["config"]["trv:0"]
+                # addr, name and model_id must be added from Shelly.GetComponents call
+                cfg.update({"addr": component["config"]["addr"]})
+                cfg.update({"name": component["config"]["name"]})
+                cfg.update(
+                    {
+                        "model_id": BLU_TRV_MODEL_ID.get(
+                            component["attrs"].get("model_id")
```

Comment:
I get `KeyError` here:
```
  File "/home/maciek/develop/aioshelly/aioshelly/rpc_device/device.py", line 573, in _retrieve_blutrv_components
    component["attrs"].get("model_id")
    ~~~~~~~~~^^^^^^^^^
KeyError: 'attrs'
```

Reply:
`attr.model_id` is under `bthomedevice:200`, not `blutrv:200`

![obraz](https://github.com/user-attachments/assets/df7bcc2a-c6e1-4843-902a-29436857ee9a)

Reply:
As you know from Teams, they are going to add it to the right key.
But I was under the impression that attrs was already there.

We can do a get for that key as well or wait for the new firmware

Reply:
We don't know when the Shelly team will deliver such a change and whether it will be part of the beta firmware that will be available for users. I think for now we should use `.get()` here to make sure our code doesn't return an exception.

Reply:
They wrote in Teams: "ok, I'll try to make the addition tomorrow and also send you the table."

Anyway, will switch to get